src/installer/corehost/cli/hostpolicy/deps_resolver.cpp

```diff
@@ -343,6 +343,12 @@ bool deps_resolver_t::probe_deps_entry(const deps_entry_t& entry, const pal::str
                 else
                 {
                     // Non-rid assets, lookup in the published dir.
```

Comment:
@vitek-karas, regarding rid-specific assets: 
* I've added the bundle-check in the rid-specific assets case too, as you've recommended.
* However, when an app is published specific to a RID, my understanding is that the SDK will never create assets for other RIDs. The SDK will find the best compatible RID assets and copy them to the publish directory.  So, there won't be a `<rid>\publish\runtimes\<rid>` structure.

For example, if we have a package (libuv 1.9.0 is a very good example)
```
runtimes
   |- debian-x64/native/lib.so
   |- rhel-x64/native/lib.so
   |- win-x64/native/lib.dll
```
`dotnet publish` will generate the above directory structure within the `bin/.net5/publish` directory.

However, `dotnet publish -r debian-x64 --self-contained=false`  will generate `bin/.net5/debian-x64/publish` that *only* contains `lib.so` from `runtime/debian-x64/native`.

`dotnet publish -r ubuntu-x64 --self-contained=false`  will generate `bin/.net5/ubuntu-x64/publish` that contains `lib.so` from `runtime/debian-x64/native`.

`dotnet publish -r linux-x64 --self-contained=false`  will generate `bin/.net5/linux-x64/publish` without `lib.so`

Reply:
> dotnet publish -r linux-x64 --self-contained=false will generate bin/.net5/linux-x64/publish without lib.so

This sounds like a problem/bug. I assume the app would not run in that case at all... right?

Reply:
I agree that there is an underlying problem, because the app may fail to run because of missing assemblies in this situation. However, I'm not sure it is an SDK bug, but rather a product issue. 

The missing RID-asset (here `linux-x64`) in a package may actually mean that the library (`lib.so`) is unnecessary for that platform -- although it is weird that the lib is not necessary at a higher RID but necessary in a lower RID. This is somewhat similar to missing an assembly that is only loaded at runtime for reflection -- where the SDK cannot know the exact requirements. 

@dsplaisted  @wli3: Should we consider guards in the SDK to check for this case? (ex: missing assets at a parent RID)? Thanks.

Reply:
@swaroop-sridhar I have a feeling trying to add guards for this type of thing would be more complex and complicated than would be warranted for the benefit we would get out of it.  But you can file an issue in the SDK repo if you want to discuss it further.

Reply:
https://github.com/dotnet/sdk/issues/11206